### PR TITLE
feat: migrate AddCategory page and its all dependencies to TS

### PR DESCRIPTION
**Pull request by Jonath-z**
# PR Description

This PR migrates the AddCategory page to TS

## Description of tasks that were expected to be completed

List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style

- [x] Migrate component to typescript
- [x] Uncomment related and file dependencies

## Screenshots (If appropriate/applicable)

Provide screenshots related to your PR in case it's appropriate and applicable

## Please check this Checklist before you submit your PR:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

### CODE REVIEW

File: components/modules/AddCategory/index.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import PropTypes from "prop-types";
```
Comment (Muhindo-Galien) [inline components/modules/AddCategory/index.tsx:2]: remove this import statement since it ain't used anymore

**Review by Muhindo-Galien — APPROVED** [review]
LGTM

**Comment by Jonath-z** [comment]
changes applied

**Comment by Muhindo-Galien** [comment]
GW

**Pull request merged**